coredns/controllers/util.go

```diff
@@ -70,14 +78,28 @@ func getCoreDNSDeployment(ctx context.Context, c client.Client) (*appsv1.Deploym
 // It is usually the 10th address to the Kubernetes Service Cluster IP
 // If the Kubernetes Service Cluster IP is not found, we default it to be "10.96.0.10"
 func findDNSClusterIP(ctx context.Context, c client.Client) (string, error) {
-	kubernetesService, err := getCoreDNSService(ctx, c)
-	if err != nil && !apierrors.IsNotFound(err) {
-		return "", err
+	kubeDNSService, err := getKubeDNSService(ctx, c)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return "", err
+		}
+	} else {
```

Comment:
I suspect you wanted to have the `else` for the inner if statement rather than the outer one.

```
if !apierrors.IsNotFound(err) {
		return "", err
} else {
```

This would fix the failing test